tests/integration/test_optislang/test_optislang_integration.py

```diff
@@ -36,69 +40,65 @@ def test_simple_solve(load_mixing_elbow_param_case_dat):
-        (Symbol("list-input-parameters"),)
-    )
-    output_parameters = solver_session.scheme_eval.eval(
-        (Symbol("list-output-parameters"),)
-    )
-    solver_session.tui.file.read_case(case_path)
-
-    # TODO: Remove the if condition after a stable version of 23.1 is available and update the commands as required.
-    if solver_session.get_fluent_version() < FluentVersion.v231:
-        input_parameters = input_parameters["inlet2_temp"]
-        output_parameters = output_parameters["outlet_temp-op"]
-
-        # Step 4: Set a variation on these input parameters
-        # variations/designs are generated by optiSLang based on
-        # algorithm selected
-        solver_session.tui.define.parameters.input_parameters.edit(
-            "inlet2_temp", "inlet2_temp", 600
-        )
-        solver_session.tui.file.write_case("design_elbow_param.cas.h5")
-
-        # Step 5: Solve
-        solver_session.tui.solve.initialize.initialize_flow()
-
-        # check if solution is steady or transient
-        workflow = solver_session.scheme_eval.string_eval("(rp-unsteady?)")
-
-        # iterate workflow
-        if workflow == "#t":
-            solver_session.tui.solve.dual_time_iterate()
-        else:
-            solver_session.tui.solve.iterate()
-
-        convergence = solver_session.scheme_eval.string_eval(
-            "(rpgetvar 'solution/converged?)"
-        )
-
-        # solution output (test conditional statement)
-        if convergence == "#f":  # -> let user know
-            print("Failed to converge")
-        else:
-            print("Solution is converged")
-
-        assert convergence == "#t", "Solution failed to converge"
-
-        # Step 6: Read the data again from the case and data file
-        solver_session.tui.file.read_case_data(case_path)
-        input_parameters2 = solver_session.scheme_eval.eval(
-            (Symbol("list-input-parameters"),)
-        )
-        output_parameters2 = solver_session.scheme_eval.eval(
-            (Symbol("list-output-parameters"),)
-        )
-        input_parameters2 = input_parameters2["inlet2_temp"]
-        output_parameters2 = output_parameters2["outlet_temp-op"]
-        assert input_parameters[0] == input_parameters2[0]
-        assert output_parameters[0] == output_parameters2[0]
-        solver_session.exit()
+    input_parameters = {}
+    for p in reader.input_parameters():
+        input_parameters[p.name] = (p.value, p.numeric_value, p.units)
+    output_parameters = {}
+    for o in reader.output_parameters():
+        output_parameters[o.name] = (0, o.units)
+    solver_session.settings.file.read_case(file_name=case_path)
+
+    input_parameters = input_parameters["inlet2_temp"]
+    output_parameters = output_parameters["outlet_temp-op"]
+
+    # Step 4: Set a variation on these input parameters
+    # variations/designs are generated by optiSLang based on
+    # algorithm selected
+    inputs_table = InputParameters(solver_session)
```

Comment:
instead of this, we can use session.parameters.input_parameters.expression[name] = {"value" : value}